### ISSUE TITLE
Leaks being detected by address santitizer

### ISSUE THREAD

**Issue by edhartnett** [NCAR/ParallelIO#1275]
```
=================================================================
==17978==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 64 byte(s) in 1 object(s) allocated from:
    #0 0x7f218aef2b50 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb50)
    #1 0x7f218635864a in NC4_buildpropinfo /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4info.c:239
    #2 0x7f21863582c9 in NC4_put_propattr /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4info.c:162
    #3 0x7f2186339f8e in nc4_create_file /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4file.c:468
    #4 0x7f218633a1b9 in NC4_create /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4file.c:564
    #5 0x7f21862af53a in NC_create /home/ed/Downloads/netcdf-c-4.4.1.1/libdispatch/dfile.c:1773
    #6 0x7f21862ac4ad in nc_create_par /home/ed/Downloads/netcdf-c-4.4.1.1/libdispatch/dparallel.c:32
    #7 0x7f218aab818c in PIOc_createfile_int ../../../../src/clib/pioc_support.c:1788
    #8 0x7f218aa53046 in PIOc_createfile ../../../../src/clib/pio_file.c:140
    #9 0x7f218aa65858 in create_file_handler ../../../../src/clib/pio_msg.c:192
    #10 0x7f218aa78ccd in pio_msg_handler2 ../../../../src/clib/pio_msg.c:2569
    #11 0x7f218aaae1e1 in PIOc_init_async ../../../../src/clib/pioc.c:1563
    #12 0x560c6b0cd630 in main ../../../../tests/cunit/test_intercomm2.c:322
    #13 0x7f2189fae1c0 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x211c0)

SUMMARY: AddressSanitizer: 64 byte(s) leaked in 1 allocation(s).

=================================================================
==17977==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 64 byte(s) in 1 object(s) allocated from:
    #0 0x7f1f30babb50 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb50)
    #1 0x7f1f2c01164a in NC4_buildpropinfo /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4info.c:239
    #2 0x7f1f2c0112c9 in NC4_put_propattr /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4info.c:162
    #3 0x7f1f2bff2f8e in nc4_create_file /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4file.c:468
    #4 0x7f1f2bff31b9 in NC4_create /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4file.c:564
    #5 0x7f1f2bf6853a in NC_create /home/ed/Downloads/netcdf-c-4.4.1.1/libdispatch/dfile.c:1773
    #6 0x7f1f2bf654ad in nc_create_par /home/ed/Downloads/netcdf-c-4.4.1.1/libdispatch/dparallel.c:32
    #7 0x7f1f3077118c in PIOc_createfile_int ../../../../src/clib/pioc_support.c:1788
    #8 0x7f1f3070c046 in PIOc_createfile ../../../../src/clib/pio_file.c:140
    #9 0x7f1f3071e858 in create_file_handler ../../../../src/clib/pio_msg.c:192
    #10 0x7f1f30731ccd in pio_msg_handler2 ../../../../src/clib/pio_msg.c:2569
    #11 0x7f1f307671e1 in PIOc_init_async ../../../../src/clib/pioc.c:1563
    #12 0x56331e5f3630 in main ../../../../tests/cunit/test_intercomm2.c:322
    #13 0x7f1f2fc671c0 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x211c0)

Direct leak of 64 byte(s) in 1 object(s) allocated from:
    #0 0x7f1f30babb50 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb50)
    #1 0x7f1f2c01164a in NC4_buildpropinfo /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4info.c:239
    #2 0x7f1f2c0112c9 in NC4_put_propattr /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4info.c:162
    #3 0x7f1f2bff2f8e in nc4_create_file /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4file.c:468
    #4 0x7f1f2bff31b9 in NC4_create /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4file.c:564
    #5 0x7f1f2bf6853a in NC_create /home/ed/Downloads/netcdf-c-4.4.1.1/libdispatch/dfile.c:1773
    #6 0x7f1f2bf6799a in nc__create /home/ed/Downloads/netcdf-c-4.4.1.1/libdispatch/dfile.c:511
    #7 0x7f1f2bf67953 in nc_create /home/ed/Downloads/netcdf-c-4.4.1.1/libdispatch/dfile.c:440
    #8 0x7f1f3077124d in PIOc_createfile_int ../../../../src/clib/pioc_support.c:1798
    #9 0x7f1f3070c046 in PIOc_createfile ../../../../src/clib/pio_file.c:140
    #10 0x7f1f3071e858 in create_file_handler ../../../../src/clib/pio_msg.c:192
    #11 0x7f1f30731ccd in pio_msg_handler2 ../../../../src/clib/pio_msg.c:2569
    #12 0x7f1f307671e1 in PIOc_init_async ../../../../src/clib/pioc.c:1563
    #13 0x56331e5f3630 in main ../../../../tests/cunit/test_intercomm2.c:322
    #14 0x7f1f2fc671c0 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x211c0)

SUMMARY: AddressSanitizer: 128 byte(s) leaked in 2 allocation(s).
running test_async_multi2
1 test_async_multi2 SUCCESS!!
5 test_async_multi2 SUCCESS!!
4 test_async_multi2 SUCCESS!!
6 test_async_multi2 SUCCESS!!
2 test_async_multi2 SUCCESS!!
7 test_async_multi2 SUCCESS!!
3 test_async_multi2 SUCCESS!!
0 test_async_multi2 SUCCESS!!

=================================================================
==18040==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 128 byte(s) in 2 object(s) allocated from:
    #0 0x7f06ead08b50 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb50)
    #1 0x7f06e616e64a in NC4_buildpropinfo /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4info.c:239
    #2 0x7f06e616e2c9 in NC4_put_propattr /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4info.c:162
    #3 0x7f06e614ff8e in nc4_create_file /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4file.c:468
    #4 0x7f06e61501b9 in NC4_create /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4file.c:564
    #5 0x7f06e60c553a in NC_create /home/ed/Downloads/netcdf-c-4.4.1.1/libdispatch/dfile.c:1773
    #6 0x7f06e60c24ad in nc_create_par /home/ed/Downloads/netcdf-c-4.4.1.1/libdispatch/dparallel.c:32
    #7 0x7f06ea8ce18c in PIOc_createfile_int ../../../../src/clib/pioc_support.c:1788
    #8 0x7f06ea869046 in PIOc_createfile ../../../../src/clib/pio_file.c:140
    #9 0x7f06ea87b858 in create_file_handler ../../../../src/clib/pio_msg.c:192
    #10 0x7f06ea88eccd in pio_msg_handler2 ../../../../src/clib/pio_msg.c:2569
    #11 0x7f06ea8c41e1 in PIOc_init_async ../../../../src/clib/pioc.c:1563
    #12 0x55f26f5aa8c1 in main ../../../../tests/cunit/test_async_multi2.c:65
    #13 0x7f06e9dc41c0 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x211c0)

Direct leak of 128 byte(s) in 2 object(s) allocated from:
    #0 0x7f06ead08b50 in __interceptor_malloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xdeb50)
    #1 0x7f06e616e64a in NC4_buildpropinfo /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4info.c:239
    #2 0x7f06e616e2c9 in NC4_put_propattr /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4info.c:162
    #3 0x7f06e614ff8e in nc4_create_file /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4file.c:468
    #4 0x7f06e61501b9 in NC4_create /home/ed/Downloads/netcdf-c-4.4.1.1/libsrc4/nc4file.c:564
    #5 0x7f06e60c553a in NC_create /home/ed/Downloads/netcdf-c-4.4.1.1/libdispatch/dfile.c:1773
    #6 0x7f06e60c499a in nc__create /home/ed/Downloads/netcdf-c-4.4.1.1/libdispatch/dfile.c:511
    #7 0x7f06e60c4953 in nc_create /home/ed/Downloads/netcdf-c-4.4.1.1/libdispatch/dfile.c:440
    #8 0x7f06ea8ce24d in PIOc_createfile_int ../../../../src/clib/pioc_support.c:1798
    #9 0x7f06ea869046 in PIOc_createfile ../../../../src/clib/pio_file.c:140
    #10 0x7f06ea87b858 in create_file_handler ../../../../src/clib/pio_msg.c:192
    #11 0x7f06ea88eccd in pio_msg_handler2 ../../../../src/clib/pio_msg.c:2569
    #12 0x7f06ea8c41e1 in PIOc_init_async ../../../../src/clib/pioc.c:1563
    #13 0x55f26f5aa8c1 in main ../../../../tests/cunit/test_async_multi2.c:65
    #14 0x7f06e9dc41c0 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x211c0)
```

**Comment by edhartnett** [comment]
I believe these are actually netCDF leaks in 4.4.1.1 that will be fixed in the upcoming 4.6.0 release. I will keep this ticket open until I confirm that they were fixed.

**Comment by edwardhartnett** [comment]
These are netcdf-c problems, and unfortunately are still not fixed. But this is not a PIO issue.
